gitlab/client.py

```diff
@@ -832,15 +835,33 @@ def http_list(
         """
         query_data = query_data or {}
 
-        # In case we want to change the default behavior at some point
-        as_list = True if as_list is None else as_list
+        # Don't allow both `as_list` and `iterator` to be set.
+        if as_list is not None and iterator is not None:
+            raise gitlab.exceptions.GitlabError(
+                error_message=(
+                    "`as_list` and `iterator` are mutually exclusive. Only one "
+                    "argument can be set. Recommend to only set `iterator` argument"
```

Comment:
No strong opinion on this, but if we stay with the messaging in the comment above and our other deprecation warnings, it would be something like:
```suggestion
                    "Only one of `as_list` and `iterator` should be defined."
                    "Use `iterator` instead of `as_list`"
```

Reply:
Done.